massa-client/src/cmds.rs

```diff
@@ -815,6 +824,66 @@ impl Command {
                 )
                 .await
             }
+            Command::call_smart_contract => {
+                if parameters.len() != 8 {
+                    bail!("wrong number of parameters");
+                }
+                let addr = parameters[0].parse::<Address>()?;
+                let target_addr = parameters[1].parse::<Address>()?;
+                let target_func = parameters[2].clone();
+                let param = parameters[3].clone();
+                let max_gas = parameters[4].parse::<u64>()?;
+                let gas_price = parameters[5].parse::<Amount>()?;
+                let coins = parameters[6].parse::<Amount>()?;
+                let fee = parameters[7].parse::<Amount>()?;
+                if !json {
+                    match gas_price
+                        .checked_mul_u64(max_gas)
+                        .and_then(|x| x.checked_add(coins))
+                        .and_then(|x| x.checked_add(fee))
+                    {
+                        Some(total) => {
+                            if let Ok(addresses_info) =
+                                client.public.get_addresses(vec![target_addr]).await
+                            {
+                                match addresses_info.get(0) {
+                                    Some(info) => {
+                                        if info.ledger_info.candidate_ledger_info.balance < total {
```

Comment:
note, in this case max_gas*gas_price+sequential_coins+fee are spent from the sequential balance, and parallel_coins are spent from the paralell balance ! so this check is invalid and  2 checks need to be performed

Reply:
Is the new check ok ?